lib/protocol/Auth.js

```diff
@@ -2,12 +2,18 @@ var Buffer = require('safe-buffer').Buffer;
 var Crypto = require('crypto');
 var Auth   = exports;
 
-function auth(name, data, options) {
+function auth(name, data, options, isSecure) {
   options = options || {};
 
   switch (name) {
     case 'mysql_native_password':
       return Auth.token(options.password, data.slice(0, 20));
+    case 'mysql_clear_password':
+      if (!isSecure) {
+        throw new Error('Authentication method mysql_clear_password not supported on insecure connections');
```

Comment:
I get where this is coming from, but is probably going to be limiting to users in the long run. What about just adding an option the user can use to enable / disable this? I'm thinking along the lines of our `insecureAuth` option (perhaps `clearAuth`?) and how the `mysql` command line program requires a flag to enable it (https://dev.mysql.com/doc/refman/8.0/en/mysql-command-options.html#option_mysql_enable-cleartext-plugin).

Reply:
It looks like the `--enable-cleartext-plugin` is just to enable the plugin itself.   Are you thinking the `insecureAuth` would allow clear_password even if we are not on a tls socket?

Reply:
I'm thinking to have it behave inline with the `mysql` command line program: add an option (like perhaps `clearAuth`?) that that user must enable to use the clear password auth. That would be the only requirement -- the user can choose to use that over SSL or not as their own choice, just like in the `mysql` command line program.

Reply:
I understand.   Looking at the python module, if you specify `auth_plugin='mysql_clear_password'` then cleartext is used whether SSL is on or not.  So maybe that would be the expected behavior.

I still think there is a big difference between saying:  I want to use cleartext only inside secure channel, vs I want to use cleartext no matter what.   For example, some of the GUI Sql clients I tested would bail on cleartext if SSL is not turned on.

Maybe the right way is to use `clearAuth = true` to enable it for secure channel, and `insecureAuth` to allow clear to be in an unencrypted socket?